jinx.el

```diff
@@ -744,6 +743,45 @@ If CHECK is non-nil, always check first."
   (modify-syntax-entry ?’ "w" jinx--syntax-table)
   (modify-syntax-entry ?. "." jinx--syntax-table))
 
+(defun jinx--correct-overlays (overlays &optional show-count)
+  "Correct words at OVERLAYS.
+If SHOW-COUNT is non-nil, show the index of the correcting words."
+  (cl-letf* (((symbol-function #'jinx--timer-handler) #'ignore) ;; Inhibit
+             (repeat-mode nil) ;; No repeating of jinx-next and jinx-previous
+             (count (length overlays))
+             (idx 0))
+    (unwind-protect
+        (while (when-let ((ov (nth idx overlays)))
+                 (let* ((deleted (not (overlay-buffer ov)))
+                        (skip
+                         (catch 'jinx--goto
+                           (unless deleted
+                             (jinx--correct
+                              (overlay-start ov)
+                              (overlay-end ov)
+                              (and show-count (format " (%d of %d)" (1+ idx) count)))))))
+                   (cond
+                    ((integerp skip) (setq idx (mod (+ idx skip) count)))
+                    ((or show-count deleted) (cl-incf idx)))))))))
+
+(defun jinx--bounds-of-word-at-point ()
+  "Return bounds of word at point as a cons cell.
+Depends of `jinx--syntax-table'."
+  (save-excursion
+    (save-match-data
+      (let* ((start (point))
+             (end (point)))
+        (set-syntax-table jinx--syntax-table)
+        (unless (looking-at-p "\\<")
+          (re-search-backward "\\<\\|^")
+          (setq start (match-beginning 0)))
+        (unless (looking-at-p "\\>")
+          (re-search-forward "\\>\\|$")
+          (setq end (match-beginning 0)))
+        (goto-char start)
+        (when (re-search-forward "\\<\\w+\\>" end t)
+          (cons (match-beginning 0) (match-end 0)))))))
```

Comment:
I am not 100% happy with that one, since it looks too long. I think this function can be simplified. Any ideas?